src/porepy/numerics/ad/operator_functions.py

```diff
@@ -109,59 +109,127 @@ def __repr__(self) -> str:
         s = f"AD Operator function '{self._name}'"
         return s
 
-    def __mul__(self, other):
-        raise RuntimeError(
-            "AD Operator functions are meant to be called, not multiplied."
+    def __neg__(self) -> Operator:
```

Comment:
Updating the arithmetic overloads, which were introduced for the class Operator.
Introducing the TypeError to avoid users operating on uncalled functions.